holmes/core/toolset_manager.py

```
@@ -15,430 +21,457 @@
-        2. Toolsets defined in self.toolsets can override both built-in and add new custom toolsets
-        3. custom toolset from config can override both built-in and add new custom toolsets # for backward compatibility
-        """
-        # Load built-in toolsets
-        builtin_toolsets = load_builtin_toolsets(dal)
-        toolsets_by_name: dict[str, Toolset] = {
-            toolset.name: toolset for toolset in builtin_toolsets
-        }
-        builtin_toolsets_names = list(toolsets_by_name.keys())
+        # Check if any toolset has invalid/unknown status
+        toolsets = data.get("toolsets", {})
+        for name, toolset_data in toolsets.items():
```

Ruff (0.12.2)

Loop control variable `name` not used within loop body

Rename unused `name` to `_name`

(B007)